### PR TITLE
fallout2-ce: init at 1.2.0

### PR DESCRIPTION
**Pull request by hughobrien**
###### Description of changes

Please see the [sister PR](https://github.com/NixOS/nixpkgs/pull/239450) as with the exception of `zlib` this is equivalent. (The two games shared largely the same engine).

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/fallout2-ce/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    rev = "v1.9";
+    hash = "sha256-/QvMQCmoocaXfDm3/c3IAPyfZqR6d7IiJ9UoFKZTpVI=";
+  };
+
+  buildInputs = [ cmake SDL2 zlib ];
+  hardeningDisable = [ "format" ];
+  preConfigure = ''
+    substituteInPlace third_party/fpattern/CMakeLists.txt --replace "FetchContent_Populate" "#FetchContent_Populate"
+    substituteInPlace third_party/fpattern/CMakeLists.txt --replace "{fpattern_SOURCE_DIR}" "${finalAttr.fpattern}"
+    substituteInPlace third_party/fpattern/CMakeLists.txt --replace "$/nix/" "/nix/"
```
Comment (SuperSandro2000) [inline pkgs/games/fallout2-ce/default.nix:32]:
```suggestion
    substituteInPlace third_party/fpattern/CMakeLists.txt \
      --replace "FetchContent_Populate" "#FetchContent_Populate" \
      --replace "{fpattern_SOURCE_DIR}" "${finalAttr.fpattern}" \
      --replace "$/nix/" "/nix/"
```
Please move this to postPatch. Can we use our system libraries?

File: pkgs/games/fallout2-ce/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  };
+
+  fpattern = fetchFromGitHub {
+    owner = "Loadmaster";
+    repo = "fpattern";
+    rev = "v1.9";
+    hash = "sha256-/QvMQCmoocaXfDm3/c3IAPyfZqR6d7IiJ9UoFKZTpVI=";
+  };
+
+  buildInputs = [ cmake SDL2 zlib ];
```
Comment (SuperSandro2000) [inline pkgs/games/fallout2-ce/default.nix:27]:
```suggestion
  nativeBuildInputs = [ cmake ];

  buildInputs = [ SDL2 zlib ];
```

File: pkgs/games/fallout2-ce/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "alexbatalov";
+    repo = "fallout2-ce";
+    rev = "v${finalAttr.version}";
+    hash = "sha256-+N4jhmxBX6z48kaU0jm90OKhguHlggT3OF9uuyY0EV0=";
+  };
+
+  fpattern = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/games/fallout2-ce/default.nix:20]: This should be in a let in. Why not package fpattern?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/games/fallout2-ce/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    owner = "alexbatalov";
+    repo = "fallout2-ce";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-+N4jhmxBX6z48kaU0jm90OKhguHlggT3OF9uuyY0EV0=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ SDL2 zlib ];
+  hardeningDisable = [ "format" ];
+  cmakeFlags = [ "-DCMAKE_BUILD_TYPE=RelWithDebInfo" ];
```
Comment (SuperSandro2000) [inline pkgs/games/fallout2-ce/default.nix:24]:
```suggestion
  cmakeBuildType = "RelWithDebInfo";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
doesn't work for me 
image

**Comment by hughobrien** [comment]
Can you try with a [known release](https://archive.org/details/Interplay_Fallout_2_Win95_2000)?

As in `fallout1-ce` the _right_ thing to do is setup the `fallout.cfg` file but the below file linking is good enough for a sanity check.

```bash
curl -LO 'https://archive.org/download/Interplay_Fallout_2_Win95_2000/Interplay-Fallout2-Win95.iso'
unar Interplay-Fallout2-Win95.iso
cd Interplay-Fallout2-Win95
rm -r master.dat data # odd indeed
ln MASTER.DAT master.dat
ln CRITTER.DAT critter.dat
ln -s DATA data
nix-shell -I nixpkgs=/src/nixpkgs -p fallout2-ce --command fallout2-ce
```
screenshot-2023-07-06T00:30:46-04:00

**Comment by SuperSandro2000** [comment]
Same as with fallout 1, I didn't do any of the setup. We should add some steps to longDescription or a link to upstream.

**Comment by NobbZ** [comment]
Same as FO-1: https://github.com/NixOS/nixpkgs/pull/239450#issuecomment-1632063160

PS: `meta.homepage` exists, isn't that a link to upstream?

**Comment by hughobrien** [comment]
Closing in favour of unified approach here https://github.com/NixOS/nixpkgs/pull/239450

**Pull request closed without merging**